### PR TITLE
DAT-11899 Create Snapshot builds of liquibase for downstream consumption

### PR DESCRIPTION
**Pull request by jnewton03**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

A clear and concise description of the change being made.  

- Introduce what was/will be done in the title
  - Titles show in release notes and search results, so make them useful
  - Use verbs at the beginning of the title, such as "fix", "implement", "improve", "update", and "add" 
  - Be specific about what was fixed or changed
  - Good Example: `Fix the --should-snapshot-data CLI parameter to be preserved when the --data-output-directory property is not specified in the command.`
  - Bad Example: `Fixed --should-snapshot-data`  
- If there is an existing issue this addresses, include "Fixes #XXXX" to auto-link the issue to this PR
- If there is NOT an existing issue, consider creating one.
  - In general, issues describe wanted change from an end-user perspective and PRs describe the technical change.
  - If this change is very small and not worth splitting off an issue, include `Steps To Reproduce`, `Expected Behavior`, and `Actual Behavior` sections in this PR as you would have in the issue.
- Describe what users need and how the fix will affect them
- Describe how the code change addresses the problem
- Ensure private information is redacted.

## Things to be aware of

- Describe the technical choices you made
- Describe impacts on the codebase

## Things to worry about

- List any questions or concerns you have with the change
- List unknowns you have

### CODE REVIEW

File: .github/workflows/build-main.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'maven'
+          server-id: github
+
+      # Version artifact based off of branch and commit SHA.
+      - name: Version Artifact
+        run: mvn versions:set -DnewVersion="$(git rev-parse --short HEAD)-SNAPSHOT"
```
Comment (StevenMassaro) [inline .github/workflows/build-main.yml:60]: You might be able to use `${{github.sha}}` here.

File: .github/workflows/build-main.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+name: Build Main/Master SNAPSHOT
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "5 6 * * *"
```
Comment (StevenMassaro) [inline .github/workflows/build-main.yml:6]: Personally, I think a nightly build is unnecessary if we're building on every push. Seems like a waste of resources to me.

**Review by StevenMassaro — APPROVED** [review]

**Review by mcred — APPROVED** [review]

**Review by abrackx — APPROVED** [review]

File: .github/workflows/build-main.yml (hunk 1 above)
Comment (jnewton03) [reply .github/workflows/build-main.yml:60]: @StevenMassaro doesn't that give the full commit has and not the short version?

File: .github/workflows/build-main.yml (hunk 2 above)
Comment (jnewton03) [reply .github/workflows/build-main.yml:6]: Easy to remove if you'd like

File: .github/workflows/build-main.yml (hunk 1 above)
Comment (StevenMassaro) [reply .github/workflows/build-main.yml:60]: Good point, it does give the full hash.

**Pull request merged**